app/code/Magento/Catalog/Model/ResourceModel/Product.php

```diff
@@ -510,6 +510,48 @@ public function getProductsIdsBySkus(array $productSkuList)
         return $result;
     }
 
+    /**
+     * Get product type by it's sku.
+     *
+     * @param string $productSku
+     * @return string
+     */
+    public function getProductTypeBySku(string $productSku): string
```

Comment:
Move to separate class in `app/code/Magento/Catalog/Model/ResourceModel/Product` namespace

Reply:
@naydav done.